### PR TITLE
Initial Vocabulary for ShapeTree

### PR DESCRIPTION
**Pull request by joshdcollins**

### CODE REVIEW

File: shapetree.ttl [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+
+
+# ShapeTree Class
+#############################################################################
+# Defines a tree of related resources which combine to enable some task
+# see: https://shapetrees.github.io/specification/primer
+#############################################################################
+:ShapeTree 
+  a rdfs:Class ;
+  rdfs:comment "Defines a tree of related resources which combine to enable some task"@en ;
```
Comment (justinwb) [inline shapetree.ttl:22]: > which combine to enable some task

Combining to enable a task feels wrong. Maybe something along the lines of (and by all means wordsmith this further)... "defines a composite data structure including a tree of resources and validation criteria"

File: shapetree.ttl [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+  rdfs:isDefinedBy <> ;
+  rdfs:label "ShapeTree"@en .
```
Comment (justinwb) [inline shapetree.ttl:24]: When writing about shape trees in regular text I typically don't combine into one word (ShapeTree), but instead do Shape Tree or shape tree depending on whether its a heading or regular text.

File: shapetree.ttl [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+
+#
+# Properties in domain of ShapeTree
+#
+:expectsType 
+  a owl:ObjectProperty ;
+  rdfs:domain :ShapeTree ;
+  rdfs:range [ owl:oneOf ( :ShapeTreeContainer :ShapeTreeResource ) ] ;
+  rdfs:comment "Describes the expected type of resource the ShapeTree will create."@en ;
```
Comment (justinwb) [inline shapetree.ttl:33]: > expected type of resource the ShapeTree will create

"create" feels a little too narrow?

File: shapetree.ttl [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+  rdfs:isDefinedBy <> ;
+  rdfs:label "Expects Type"@en .
+
+:contains 
+  a owl:ObjectProperty ;
+  rdfs:domain :ShapeTree ;
+  rdfs:range :ShapeTree  ;
+  rdfs:comment "Links a ShapeTree to another ShapeTree that is directly contained within its planted container"@en ;
```
Comment (justinwb) [inline shapetree.ttl:41]: maybe swap "container" for "ShapeTreeContainer"

File: shapetree.ttl [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+# Specific IRIs used by :contains predicate to define behavior 
+# when a matching ShapeTree with a URI Template is not found.
+# These values can be combined, for example,
+# :contains :AllowResources :AllowContainers :AllowNonRDFSources is equivalent to :AllowAll
+#
+:AllowResources
+  a :ShapeTree ;
+  rdfs:comment "If no matching :contains ShapeTree is found, allow if the requested content is a :ShapeTreeResource"@en ;
+  rdfs:isDefinedBy <> ;
+  rdfs:label "AllowResources"@en .
```
Comment (justinwb) [inline shapetree.ttl:86]: Maybe expand these Allow* rdfs labels (e.g. "Allow Resources", "Allow Containers")

File: shapetree.ttl [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+  rdfs:isDefinedBy <> ;
+  rdfs:label "ShapeTreeContainer"@en .
+
+
+# ShapeTreeResource IRI (maps to ldp:Resource)
+#############################################################################
+# An IRI that represents a generalized resource used in a ShapeTree
+#############################################################################
+:ShapeTreeResource 
+  rdfs:comment "Describes a generalized resource (non-container) within a ShapeTree ecosystem"@en ;
```
Comment (justinwb) [inline shapetree.ttl:127]: comment may want to call it out as an rdf resource if everything else is a "NonRDFSource".

File: shapetree.ttl [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+  rdfs:range xsd:string ;
+  rdfs:comment "Describes the version of a ShapeTreeDecoratorHierarchy"@en ;
+  rdfs:isDefinedBy <> ;
+  rdfs:label "Has Version"@en .
+
+:hasSHA3256Hash
+  a owl:DatatypeProperty ;
+  rdfs:domain :ShapeTreeDecoratorHierarchy ;
+  rdfs:range xsd:string ;
+  rdfs:comment "Describes the SHA3-256 of a ShapeTreeDecoratorHierarchy"@en ;
```
Comment (justinwb) [inline shapetree.ttl:244]: Given some of the feedback we got on using SHA3 in ecosystem proposal would SHA-256 be sufficient?

File: shapetree.ttl [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+  rdfs:label "Has Root ShapeTree"@en .
+
+
+################################################################
+# A generalized predicate for linking a class to a ShapeTree
+################################################################
+:hasShapeTree
+  a owl:ObjectProperty ;
+  rdfs:range :ShapeTree ;
+  rdfs:comment "Represents the IRI of a ShapeTree that "@en ;
```
Comment (justinwb) [inline shapetree.ttl:323]: Should this comment be cut off?

File: shapetree.ttl [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+  rdfs:range :ShapeTree ;
+  rdfs:comment "Describes a relationship wherein a ShapeTree acts as a support for another, for purposes of indexing, aggregation, etc." ;
+  rdfs:isDefinedBy <> ;
+  rdfs:label "Supports"@en .
+
+#
+# Specific IRIs used by :contains predicate to define behavior 
+# when a matching ShapeTree with a URI Template is not found.
+# These values can be combined, for example,
+# :contains :AllowResources :AllowContainers :AllowNonRDFSources is equivalent to :AllowAll
```
[the remark this answers is not in the document]
Comment (joshdcollins) [reply shapetree.ttl:80]: Updated.

File: shapetree.ttl (hunk 1 above)
Comment (joshdcollins) [reply shapetree.ttl]: Updated.

File: shapetree.ttl (hunk 2 above)
Comment (joshdcollins) [reply shapetree.ttl]: Updated.

File: shapetree.ttl (hunk 3 above)
Comment (joshdcollins) [reply shapetree.ttl]: Updated.

File: shapetree.ttl (hunk 4 above)
Comment (joshdcollins) [reply shapetree.ttl]: Updated.

File: shapetree.ttl (hunk 5 above)
Comment (joshdcollins) [reply shapetree.ttl]: Updated.

File: shapetree.ttl (hunk 6 above)
Comment (joshdcollins) [reply shapetree.ttl]: Updated.

File: shapetree.ttl (hunk 7 above)
Comment (joshdcollins) [reply shapetree.ttl]: Updated.

File: shapetree.ttl (hunk 8 above)
Comment (joshdcollins) [reply shapetree.ttl]: Updated.

File: shapetree.ttl (hunk 7 above)
Comment (justinwb) [reply shapetree.ttl]: @joshdcollins Comment and label looks good but the actual property name is still `:hasSHA3256Hash` 🙂

Comment (joshdcollins) [reply shapetree.ttl]: Updated for real.

**Pull request merged**